class/catalog.class.php

```diff
@@ -836,7 +913,441 @@ public function GetFilecsv()
 			$response = '0,<p>Aucune mise à jour catalogue Ecopresto disponible.</p>';
 		return $response;
 	}
+	/**
+	 * Enregistre le catalogue CSV en provennance d'Ecopresto
+	 *
+	 *
+	 * @param aucun
+	 * @return false si erreur, nb ligne du fichier sinon
+	 */
+	public function GetCatalogCSV()
+	{
+		$domain = Configuration::get('PS_SHOP_DOMAIN');
+		
+		$cle = $this->tabConfig['ID_ECOPRESTO'];
+
+		$filename = $this->fichierDistant;
+		include_once _PS_ROOT_DIR_.'/modules/ecopresto/class/download.class.php';
+		$download = new DownloadBinaryFile();
+		$response = '';
+		
+		
+		if ($download->load($filename) == true)
+		{
+			$download->saveTo(_PS_ROOT_DIR_.'/modules/ecopresto/files/'.$this->fichierImport);

```

Comment:
This variable has the value *'catalogue.csv'*. It would be good to generate a name to avoid people accessing this file by writing the URL directly.